src/k8s/cmd/k8s/main.go

```diff
@@ -1,11 +1,14 @@
 package k8s
 
-import "os"
+import (
+	"fmt"
+	"os"
+)
 
 func Main() {
-	if rootCmd.Execute() != nil {
-		// TODO: We need to define actionable error message in the future
-		// That tell the user what went wrong on a high-level and - if possible - how it can be fixed.
+	rootCmd := NewRootCmd()
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Fprintln(rootCmd.ErrOrStderr(), err)
```

Comment:
could it be simpler if we just handle the error here? i'd rather not introduce an error handler that we have to remember to use all our commands with.

It should be enough to have the command's `RunE()` return the error, then call the logic of `resolveError()` here.

Reply:
Also, the client should not be initialized during `init()`, but rather as a step for each command. It's okay to keep some _cursed_ global state for this, and have a ConsistentPreRun on commands that need to initialize the client. Let's consider how we can inject a mock at a later time. If it's one place that initializes the client, it will be easy to do.

Reply:
The client creation also blocks the e2e tests atm, and the error there makes sense (avoid initializing _runtime_ things that we might not even need in all commands during `init()`)